src/Evolution/Systems/GrMhd/GhValenciaDivClean/TimeDerivativeTerms.hpp

```diff
@@ -0,0 +1,244 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/System.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/TimeDerivative.hpp"
+#include "Evolution/Systems/GrMhd/GhValenciaDivClean/StressEnergy.hpp"
+#include "Evolution/Systems/GrMhd/GhValenciaDivClean/Tags.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/TimeDerivativeTerms.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace grmhd::GhValenciaDivClean {
+namespace detail {
+template <typename GhDtTagList, typename ValenciaDtTagList,
+          typename ValenciaFluxTagList, typename GhTempTagList,
+          typename ValenciaTempTagList, typename GhGradientTagList,
+          typename GhArgTagList, typename ValenciaArgTagList>
+struct TimeDerivativeTermsImpl;
+
+template <typename... GhDtTags, typename... ValenciaDtTags,
+          typename... ValenciaFluxTags, typename... GhTempTags,
+          typename... ValenciaTempTags, typename... GhGradientTags,
+          typename... GhArgTags, typename... ValenciaArgTags>
+struct TimeDerivativeTermsImpl<
+    tmpl::list<GhDtTags...>, tmpl::list<ValenciaDtTags...>,
+    tmpl::list<ValenciaFluxTags...>, tmpl::list<GhTempTags...>,
+    tmpl::list<ValenciaTempTags...>, tmpl::list<GhGradientTags...>,
+    tmpl::list<GhArgTags...>, tmpl::list<ValenciaArgTags...>> {
+  static void apply(
+      const gsl::not_null<typename GhDtTags::type*>... gh_dts,
+      const gsl::not_null<typename ValenciaDtTags::type*>... valencia_dts,
+      const gsl::not_null<typename ValenciaFluxTags::type*>... valencia_fluxes,
+      const gsl::not_null<typename GhTempTags::type*>... gh_temporaries,
+      const gsl::not_null<
+          typename ValenciaTempTags::type*>... valencia_temporaries,
+      gsl::not_null<tnsr::aa<DataVector, 3_st>*> stress_energy,
+      gsl::not_null<tnsr::a<DataVector, 3_st>*> four_velocity_one_form,
+      gsl::not_null<tnsr::a<DataVector, 3_st>*>
+          comoving_magnetic_field_one_form,
+      const typename ::Tags::deriv<GhGradientTags, tmpl::size_t<3_st>,
+                                   Frame::Inertial>::type&... gh_gradients,
+      const typename GhArgTags::type&... gh_args,
+      const typename ValenciaArgTags::type&... valencia_args) noexcept {
+    grmhd::ValenciaDivClean::TimeDerivativeTerms::apply(
```

Comment:
The ordering here is not what I expected. I thought this would call:
1. GH time derivative, which would compute lapse shift, etc. that are all needed by MHD
2. MHD time derivative which might compute things for the stress tensor
3. compute and add stress tensor to `Pi` equation

How can this work with the ordering present here? Is it relying on a bunch of compute tags or something? If so, I'd like to avoid that because compute tags will be somewhere between completely incompatible and a total nightmare when it comes to subcell. I think the combined system should really only take the evolved vars, prims, H_a, and the damping functions. Everything else should be computed internally to make it have the best chance of being reasonably subcell-compatible.

Reply:
I can certainly reorder in the way you suggest, but I'm confused about the central point of the question.
In writing this, I've assumed that the `ValenciaDivClean` time derivative calculation is already 'self-sufficient' in the sense that it either already computes internally all of the spacetime quantities that it needs, or that they are already contained elsewhere.
Is the `ValenciaDivClean` time derivative calculation currently just assuming a static spacetime and taking everything computed once from the analytic solution initialized at the start? If so, then I agree that this needs to be reordered.

Reply:
Ah, yea, so `ValenciaDivClean` current computes the spacetime quantities via compute tags from the analytic solution (this is already really annoying for subcell because compute tags assume a specific grid, etc. and so in that sense GH+GRMHD can actually be easier). From the subcell perspective, it is by far the easiest if everything is not computed via compute tags, or at least as few things as absolutely possible. What that means for the combined system is the order I had above with the `argument_tags` not just being the `tmpl::append<GH,GRMHD>` but actually a more curated list would make life quite a bit easier. 

Having said all that, subcell will probably still be tough to get working in either case because with subcell the source terms and the nonconservative products in GH are computed at the cell centers while the fluxes are needed at the cell faces. So my statement is this: "It would be nicer if it's not too difficult to avoid relying on the spacetime quantities like lapse, shift, etc. being retrievable from the DataBox and instead be retrieved from the temp tags. If this is a big pain, we can defer and see if it turns out to actually be a blocker." Does that sound okay?

Reply:
Aha, I see -- so I think we can omit the compute tags for the GH+GRMHD system, that seems feasible.
Let me see if I understand the nature of the design -- is it true that we can assume that all of the spacetime quantities that `ValenciaDivClean` needs are specified in the `argument_tags`?
Then, the hope is to just use those `temporary_tags` from GH that are directly needed by `ValenciaDivClean`, and insert additional explicit helpers function calls to compute the tags that are needed by `ValenciaDivClean` but not computed by GH.

Reply:
Yes, that's right! I think that would actually work well. Basically, `tmpl::filter` out the GH temporary tags from the `DivClean` `argument_tags`? I like that idea. There will be a few extra argument tags in `DivClean` like the divergence cleaning parameter and the EOS, but those won't be filtered out in that case :)